### DISCUSSION TITLE
When to choose action over api?

### DISCUSSION THREAD

**Discussion by alamenai** [vercel/next.js#68105] [Help]
### Summary

Hi everyone,

Before the creation of action, we were relying on the api folder for handling the API requests, however recently I find myself doing everything in actions even I don't have forms (Our  API was built in python).

I would understand the limitation and the use case of Actions  because I became addictive to them!

**Comment by icyJoseph** [comment]
I'd go for an API route, or Route Handler, if I need to look at more stuff from the request, and return other data types, like download a file, etc ~ Also in general if you want to be able to reach the endpoint through other means, not just through your UI.

Would like to hear what others think here.
